Revert "Lookup Table Limit Restore to Recently Modified"

AmitPhulera opened the pull request:
Creating this PR just in case we were able to confirm that the current issues in fixtures are because of this PR - 
Reverts dimagi/commcare-hq#36091

AmitPhulera committed:
Revert "Lookup Table Limit Restore to Recently Modified"

millerdev commented:
Noting the revert instructions from the reverted PR:

> This PR includes a migration that adds `last_modified` fields for `LookupTable`. If this PR/the migration is reverted, we have to ensure no other changes have been made that uses this field.

It may be cleaner to revert specific parts of that PR (excluding the migration and any code needed to keep `LookupTable.last_modified` up to date).

Jtang-1 committed:
keep changes involving last_modified migration and updating its value

Jtang-1 committed:
query metadb by key. This is because we previously saved multiple blobs with the same parent_id, type_code, and name
but with different keys. Querying by parent_id, type_code, and name was resulting in getting the wrong blob and resultingly, the wrong key

key is indexed so would not affect performance
